src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,174 @@
+        findSectionBySameDownStation(downStation)
+                .ifPresent(section -> section.updateDownStation(upStation, distance));
+    }
+
+    private void updateUpStation(Station upStation, Station downStation, int distance) {
+        findSectionBySameUpStation(upStation)
+                .ifPresent(section -> section.updateUpStation(downStation, distance));
+    }
+
+    private Optional<Section> findSectionBySameUpStation(Station station) {
+        return sections.stream()
+                .filter(it -> it.isSameWithUpStation(station))
+                .findFirst();
+    }
+
+    private Optional<Section> findSectionBySameDownStation(Station station) {
+        return sections.stream()
+                .filter(it -> it.isSameWithDownStation(station))
+                .findFirst();
+    }
+
+    private Section getSectionBySameUpStation(Station station) {
+        return sections.stream()
+                .filter(it -> it.isSameWithUpStation(station))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(SECTION_IS_NOT_CONTAIN_STATION.message()));
+    }
+
+    private Section getSectionBySameDownStation(Station station) {
+        return sections.stream()
+                .filter(it -> it.isSameWithDownStation(station))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(SECTION_IS_NOT_CONTAIN_STATION.message()));
+    }
+
+    private boolean isExistStation(Station station) {
+        return getStations().stream().anyMatch(it -> it.isSame(station));
+    }
+
+    private void validAddableSection(boolean isUpStationExisted, boolean isDownStationExisted) {
+        validAddSameSection(isUpStationExisted, isDownStationExisted);
+        validReachableSection(isUpStationExisted, isDownStationExisted);
+    }
+
+    private void validAddSameSection(boolean isUpStationExisted, boolean isDownStationExisted) {
+        if (isUpStationExisted && isDownStationExisted) {
+            throw new IllegalArgumentException(SECTION_ALREADY_REGISTERED.message());
+        }
+    }
+
+    private void validReachableSection(boolean isUpStationExisted, boolean isDownStationExisted) {
+        if (!getStations().isEmpty() && !isUpStationExisted && !isDownStationExisted) {
+            throw new IllegalStateException(SECTION_NOT_REACHABLE_ANY_STATION.message());
+        }
+    }
+
+    private void validRemovableSection(Station station) {
+        validSectionsSize();
+        validExistStation(station);
+    }
+
+    private void validExistStation(Station station) {
+        if (!isExistStation(station)) {
+            throw new IllegalStateException(SECTION_NOT_REACHABLE_ANY_STATION.message());
+        }
+    }
+
+    private void validSectionsSize() {
+        if (sections.size() <= 1) {
+            throw new IllegalStateException(LINE_MIN_SECTIONS_SIZE.message());
+        }
```

Comment:
```suggestion
        if (sections.size() <= MIN_SIZE) {
            throw new IllegalStateException(LINE_MIN_SECTIONS_SIZE.message());
        }
```
매직넘버에 이름을 부여하는건 어떨까요? 😀

Reply:
사소한 것이지만, 익숙하지 않아서 그런지 자꾸 놓치게 되네요😭
꼼꼼하게 봐주셔서 감사합니다. 다음 단계진행 시 바로 반영하겠습니다!